docs/_source/docs/hooks.rst

```diff
@@ -44,23 +44,105 @@ Available Hooks
 cmd
 ~~~
 
-Executes the argument string in the shell as a Python subprocess.
-
-For more information about how this works, see the `subprocess documentation`_
+Executes a command through the shell.
 
 Syntax:
 
 .. code-block:: yaml
 
+   # Default shell.
    <hook_point>:
      - !cmd <shell_command>
 
-Example:
+   # Another shell.
+   <hook_point>:
+     - !cmd
+         run: <shell_command>
+         shell: <shell_name_or_path>
+
+Pass the command string as the only argument to use the default shell.
+
+On POSIX the default shell is ``sh``. On Windows it's usually ``cmd.exe``, but ``%ComSpec%`` may
+override that.
+
+Write the command string as you would type it at the shell prompt. This includes quotes and
+backslashes to escape filenames with spaces in them. To minimize escaping, you can use YAML plain
+strings like the following examples.
+
+.. code-block:: yaml
+
+   hooks:
+     before_update:
+       - !cmd echo "Hello, world!"
+
+A successful command prints its standard output messages between status messages from Sceptre.
+
+.. code-block::
+
+   [2023-09-03 01:06:28] - test - Launching Stack
+   [2023-09-03 01:06:29] - test - Stack is in the CREATE_COMPLETE state
+   Hello, world!
+   [2023-09-03 01:06:31] - test - Updating Stack
+   [2023-09-03 01:06:31] - test - No updates to perform.
+
+Pass named arguments to use a different shell. Here the command string is called ``run`` and the
+shell executable is called ``shell``.
+
+Write the executable name as you would type it at an interactive prompt to start the shell. For
+example, if Bash is in the system path, you can write ``bash``; otherwise, you need to write the
+absolute path such as ``/bin/bash``.
+
+.. code-block:: yaml
+
+   hooks:
+     before_update:
+       - !cmd
+           run: echo "Hello, $0!"
```

Comment:
this parameter change is not backwards compatible.  I think this change should be in a separate PR with a separate discussion thread

Reply:
Can you tell me what "backwards compatible" means for this feature?

The latest released version of Sceptre is v4.2.2 and supports only the string argument. The keyword arguments, whatever we decide to call them, are a new feature. I thought that as long as we continue to support the string argument then we satisfy the backwards-compatible requirement.

Over the weekend I renamed `command` to `run`, but only in the branch.

Reply:
oops, I misread.  i thought you wanted to replace `!cmd` with `!run`.  sorry about that.  replacing `command` is fine since it's never been used before.

Reply:
Ah, okay. Yes, there's no need to change the name of the hook. Apparently that's controlled in poetry.toml.

https://github.com/Sceptre/sceptre/blob/33404ee5486ff345ed738044aeb1cc3b71fdb129/pyproject.toml#L30